### PR TITLE
WIP: Update "Futures and Promises" Japanese documentation

### PR DESCRIPTION
**Pull request by seratch**
This pull request contains the following changes to fix some outdated parts of ["Futures and Promises" Japanese documentation](http://docs.scala-lang.org/ja/overviews/core/futures.html).
- Add Execution Context
- Update some outdated lines
- Replace deprecated future method with Future.apply

### CODE REVIEW

**Comment by heathermiller** [comment]
Thanks! @eed3si9n, would you care to do a quick pass to double check for errors? (It's short!)

**Comment by eed3si9n** [comment]
It would be easier to review tracking changes if the commit message included the original change commits like this - 43e4f9110b6b1f248424d9cb3f87413a11897deb

File: ja/overviews/core/futures.md [hunk 1]
```diff
@@ -99,7 +276,7 @@ Future オブジェクトを返すメソッドだということだ。
 テキストファイルがあったとして、その中である特定のキーワードが最初に出てきた位置を知りたいとする。
 この計算はファイルの内容をディスクから読み込むのにブロッキングする可能性があるため、他の計算と並行実行するのは理にかなっている。
 
-    val firstOccurence: Future[Int] = future {
+    val firstOccurence: Future[Int] = Future {
```
Comment (eed3si9n) [inline ja/overviews/core/futures.md]: s/firstOccurence/firstOccurrence/ tracking ce854132ff1bb0a13e7cbe760add4d8bcaf23802

File: ja/overviews/core/futures.md [hunk 2]
```diff
@@ -179,7 +356,7 @@ Future の実装の多くは、Future の結果を知りたくなったクライ
 
 キーワードの初出の位置を検索する例に戻ると、キーワードの位置を画面に表示したいかもしれない:
 
-    val firstOccurence: Future[Int] = future {
+    val firstOccurence: Future[Int] = Future {
```
Comment (eed3si9n) [inline ja/overviews/core/futures.md]: s/firstOccurence/firstOccurrence/ tracking ce854132ff1bb0a13e7cbe760add4d8bcaf23802

**Comment by eed3si9n** [comment]
Note: 1b13478aff1879719da47781a559d05bb69564df tracking was skipped at「 次に、仮想の `createSessionFor` メソッドを呼んでサーバにリクエストを送るセッション変数を初期化している。」
should be 「次に、`List[Friend]` を返す仮想の `createSessionFor` メソッドを呼んでサーバにリクエストを送るセッション変数を初期化している。」

(Not sure if we need to hold Sera-san responsible for getting everything up to date)

**Comment by heathermiller** [comment]
@seratch do you think you'd be able to make @eed3si9n's suggested changes? Should be quick.

Thank you @eed3si9n for the fast turnaround! It's awesome!

File: ja/overviews/core/futures.md [hunk 3]
```diff
@@ -26,19 +26,196 @@ The futures and promises API builds upon the notion of an
...
+
+    implicit val ec: ExecutionContext = ...
+    val inverseFuture : Future[Matrix] = Future {
+      fatMatrix.inverse()
+    } // ec is implicitly passed
+
+<!-- Both code snippets delegate the execution of `fatMatrix.inverse()` to an `ExecutionContext` and embody the result of the computation in `inverseFuture`. -->
+どちらのコードスニペットも `fatMatrix.inverse()` の実行を `ExecutionContext` に移譲し、その計算結果を `inverseFuture` として具体化している。
+
+## Execution Context
```
Comment (eed3si9n) [inline ja/overviews/core/futures.md]: 原文: Execution Context
訳案: 実行コンテキスト

後の方で、「実行コンテキスト（execution context）」という記述がありますが、それをやるならここで訳語を見出しに持ってきたほうがいいと思います。もしくは、ずっと Execution Context で通すか。

File: ja/overviews/core/futures.md [hunk 4]
```diff
@@ -26,19 +26,196 @@ The futures and promises API builds upon the notion of an
...
+(although executing the computation in the current thread is discouraged -- more on that below).-->
+`ExecutionContext` は [Executor](http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Executor.html) のようなもので、計算の実行は新しく生成したスレッド、コネクションプールで管理されるスレッド、実行中のスレッド、どのスレッドで実行しようと自由だ（ただし、詳細は後述するが実行中のスレッドでそのまま計算を実行するのは推奨されない）。
+
+<!--The `scala.concurrent` package comes out of the box with an `ExecutionContext` implementation, a global static thread pool.
+It is also possible to convert an `Executor` into an `ExecutionContext`.
+Finally, users are free to extend the `ExecutionContext` trait to implement their own execution contexts,
+although this should only be done in rare cases.-->
+`scala.concurrent` パッケージでは、すぐに使える `ExecutionContext` の実装を提供していて、これはグローバルで静的なスレッドプールを使って動作するものだ。`Executor` を `ExecutionContext` に変換することもできる。また、これはまれなケースではあるものの、最終的には利用者が `ExecutionContext` trait を継承した自前の実装を用いることも可能だ。
+
+### The Global Execution Context
```
Comment (eed3si9n) [inline ja/overviews/core/futures.md]: 原文: The Global Execution Context
訳案1: グローバルな実行コンテキスト
訳案2: グローバルな Execution Context

「グローバルな」という言い方は本文中でしているので、この見出しは和訳可能だと思います。

File: ja/overviews/core/futures.md [hunk 5]
```diff
@@ -26,19 +26,196 @@ The futures and promises API builds upon the notion of an
...
+        }
+      }
+    }
+
+
+<!--If you need to wrap long lasting blocking operations we recommend using a dedicated `ExecutionContext`, for instance by wrapping a Java `Executor`.-->
+もしあなたが長時間続くブロッキングな処理を Future に包む必要があるなら、例えば Java の `Executor` をラップするなどして専用の `ExecutionContext` を使うことを薦める。
+
+
+### Adapting a Java Executor
```
Comment (eed3si9n) [inline ja/overviews/core/futures.md]: 原文: Adapting a Java Executor
訳案: Java Executor からの適合

File: ja/overviews/core/futures.md [hunk 6]
```diff
@@ -26,19 +26,196 @@ The futures and promises API builds upon the notion of an
...
+
+<!--Using the `ExecutionContext.fromExecutor` method you can wrap a Java `Executor` into an `ExecutionContext`.
+For instance:-->
+
+`ExecutionContext.fromExecutor` メソッドを使えば Java の `Executor` を `ExecutionContext` にすることができる。以下に使用例を示す。
+
+    ExecutionContext.fromExecutor(new ThreadPoolExecutor( /* your configuration */ ))
+
+
+### Synchronous Execution Context
```
Comment (eed3si9n) [inline ja/overviews/core/futures.md]: 原文: Synchronous Execution Context
訳案: 同期実行コンテキスト または 同期 Execution Context

File: ja/overviews/core/futures.md [hunk 7]
```diff
@@ -26,19 +26,196 @@ The futures and promises API builds upon the notion of an
...
+<!--
+As stated above the `ForkJoinPool` can increase the amount of threads beyond its `parallelismLevel` in the presence of blocking computation.
+As explained in the `ForkJoinPool` API, this is only possible if the pool is explicitly notified:-->
+上記で述べた通り、`ForkJoinPool` はブロッキング演算が存在する場合にその並列度に応じてスレッド数を増やすことがある。
+ただし、`ForkJoinPool` の API で説明されているように、それを明示的にスレッドプールに伝えて初めて可能となる。
+
+    import scala.concurrent.Future
+    import scala.concurrent.forkjoin._
+
+    // the following is equivalent to `implicit val ec = ExecutionContext.global`
```
Comment (eed3si9n) [inline ja/overviews/core/futures.md]: 訳し抜け。

File: ja/overviews/core/futures.md [hunk 8]
```diff
@@ -26,19 +26,196 @@ The futures and promises API builds upon the notion of an
...
+実行中スレッドで計算を実行する `ExecutionContext` をつくりたい誘惑にかられるかもしれない。
+
+    val currentThreadExecutionContext = ExecutionContext.fromExecutor(
+      new Executor {
+        // Do not do this!
+        def execute(runnable: Runnable) { runnable.run() }
+    })
+
+<!--This should be avoided as it introduces non-determinism in the execution of your future.-->
+これは Future の実行に不確定性をもたらしてしまうので避けるべきだ。
```
Comment (eed3si9n) [inline ja/overviews/core/futures.md]: 原文: non-determinism
訳案: 非決定性

https://ja.wikipedia.org/wiki/%E9%9D%9E%E6%B1%BA%E5%AE%9A%E6%80%A7%E6%9C%89%E9%99%90%E3%82%AA%E3%83%BC%E3%83%88%E3%83%9E%E3%83%88%E3%83%B3

File: ja/overviews/core/futures.md [hunk 9]
```diff
@@ -26,19 +26,196 @@ The futures and promises API builds upon the notion of an
...
+
+`ExecutionContext.fromExecutor` メソッドを使えば Java の `Executor` を `ExecutionContext` にすることができる。以下に使用例を示す。
+
+    ExecutionContext.fromExecutor(new ThreadPoolExecutor( /* your configuration */ ))
+
+
+### Synchronous Execution Context
+
+<!--One might be tempted to have an `ExecutionContext` that runs computations within the current thread:-->
+実行中スレッドで計算を実行する `ExecutionContext` をつくりたい誘惑にかられるかもしれない。
```
Comment (eed3si9n) [inline ja/overviews/core/futures.md]: 原文: the current thread
訳案: 現在実行中のスレッド

**Comment by seratch** [comment]
@heathermiller @eed3si9n Thank you for your quick response! 

>It would be easier to review tracking changes if the commit message included the original change commits like this - 43e4f91

Got it, I'll do that. I'll fix all the untracked changes as well.

**Comment by eed3si9n** [comment]
> Got it, I'll do that. I'll fix all the untracked changes as well.

Thanks! Sorry for pulling you into the rabbit hole :)

**Comment by heathermiller** [comment]
Thank you @seratch!

**Comment by SethTisue** [comment]
@seratch @eed3si9n want to get this across the finish line...?

**Comment by seratch** [comment]
@SethTisue Sorry for my late reply. I couldn't work on this after that... When I will start working on this, I will do a rebase with the latest revision. I've marked the subject of this PR as WIP so far.

**Comment by SethTisue** [comment]
@seratch want to get this over the finish line...?

**Comment by seratch** [comment]
@SethTisue At the current moment, I'd like to spend my time for other things. I hope someone will take over the work. Anyway, let me close this so far.

**Pull request closed without merging**